---
title: "composer/ERR/2023_002"
linkTitle: "ERR/2023_002"
weight: 1
type: docs
description: >
 Cloud Composer private IP Cluster non-RFC1918 IP range
---

**Product**: [Cloud Composer](https://cloud.google.com/composer)\
**Rule class**: ERR - Something that is very likely to be wrong

### Description

Private IP cluster (Pods, Services) Should use ALLOWED IP RANGES
to create the environment.Make sure you are using ALLOWED IP RANGES
during environment Creation.

### Remediation

  Use Private IP Range or Non-RFC1918 AllOWED IP RANGES to
  create private IP cluster in Composer Environment.

  ALLOWED_IP_RANGES :

  Private IP Range
  ```
     ['10.0.0.0/8','172.16.0.0/12','192.168.0.0/16']
  ```

  Non-RFC1918 IP Range
  ```
    ['100.64.0.0/10','192.0.0.0/24','192.0.2.0/24','192.88.99.0/24','198.18.0.0/15','198.51.100.0/24','203.0.113.0/24','240.0.0.0/4' ]
  ```

### Further information

- [Non-RFC 1918 address ranges are partially supported for Pods and Services](https://cloud.google.com/composer/docs/known-issues#non-rfc_1918_address_ranges_are_partially_supported_for_pods_and_services)
- [Valid IPv4 ranges](https://cloud.google.com/vpc/docs/subnets#valid-ranges)
