---
title: "gke/ERR/2023_007"
linkTitle: "ERR/2023_007"
weight: 1
type: docs
description: >
  GKE Gateway controller reporting missing or invalid resource references in Gateway resource
---

**Product**: [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine)\
**Rule class**: ERR - Something that is very likely to be wrong

### Description

Gateway controller creates urls maps based on HTTPRoute resources.
If the HTTP route has invalid reference to gateway or invalid spec,
controller will not be able create url maps.

You can use the following filter to find matching log entries with gateway resource
containing misconfigured httproute:

```
jsonPayload.MESSAGE =~"Error GWCER106: Gateway"
resource.type="k8s_cluster"
```

### Remediation

- Fix the HTTProute resource

### Further information
