---
title: "gcf/ERR/2022_003"
linkTitle: "ERR/2022_003"
weight: 1
type: docs
description: >
  Function invocation interrupted due to memory limit exceeded.
---

**Product**: [Cloud Functions](https://cloud.google.com/functions)\
**Rule class**: ERR - Something that is very likely to be wrong

### Description

When configuring Cloud Functions, you can choose how much memory is allocated to run the function. By default, the memory allocated to each function is 256 MB. You should ensure that the function executes within the amount of memory specified otherwise they will encounter an Out-of-Memory (OOM) error which will interrupt their function invocation.

### Remediation

This will vary depending on runtime, code logic and the use-case(s) employed by you.

1. Inspect your peak memory utilisation (P99) using Metrics Explorer by configuring a graph showing [user memory bytes](https://cloud.google.com/monitoring/api/metrics_gcp#gcp-cloudfunctions) for the specific function. Please set the [Aggregator field](https://cloud.google.com/monitoring/charts/metrics-selector#groupby-option) to 99th percentile to display P99 data. Look for patterns and trends where peak memory utilisation is (consistently) showing peak memory usage near the memory allocated limit If so, we encourage you to always refer to [best practices](https://cloud.google.com/functions/docs/bestpractices/tips#correctness) to ensure they are inline with our recommendations. You should review the function code to check for any memory leaks as per following:
  -  Always delete temporary files
     - check if you are using any temporary files in your code as they sometimes persist between invocations. As per this [Always delete temporary files](https://cloud.google.com/functions/docs/bestpractices/tips#always_delete_temporary_files)   files that are written consume memory available for the function and sometimes persist between function invocations. Failing to explicitly delete these files may eventually lead to an OOM error and a subsequent cold start.
  -  Check resources in local and global scope
     -  If possible, do not store any objects or variables in global memory space without deallocating them as they will persist in the server instance indefinitely or until the instance is shut down. If implemented, customers should also do lazy initialisation of global variables to ensure resources are only allocated if/when their function executes.
  -  Increase code verbosity
     - Increase code verbosity by writing sufficient debugging [logs](https://cloud.google.com/functions/docs/monitoring/logging) to determine causes of high memory usage in their code.
  -  Implement memory profiling
     - Instrument your application by implementing memory profiling for individual functions to better understand their memory usage. For example, if you are deploying Python runtime you may consider using open-source libraries such as tracemalloc or guppy3.

2. If there is a pressing need and as a last resort option, you are adviced to allocate (higher) memory in accordance with analysis from inspecting their P99 memory usage using the below command:

```
gcloud functions deploy [*FUNCTION_NAME*] --memory=[*MEMORY*]
```


### Further information

#### Symptoms
- **Error**: ```Function invocation was interrupted. Error: memory limit exceeded.```

To view this error in Cloud Logging, please refer to the following query:


```
resource.type = "cloud_function"
resource.labels.function_name = "[*FUNCTION_NAME*]"
severity>=DEFAULT
textPayload: "Error: memory limit exceeded."
```

#### References:

- [Cloud Functions execution environment](https://cloud.google.com/functions/docs/concepts/exec#memory)

- [Metrics from Cloud Functions](https://cloud.google.com/monitoring/api/metrics_gcp#gcp-cloudfunctions)

- [Group time series](https://cloud.google.com/monitoring/charts/metrics-selector#groupby-option)

- [best practices](https://cloud.google.com/functions/docs/bestpractices/tips#correctness)

- [Always delete temporary files](https://cloud.google.com/functions/docs/bestpractices/tips#always_delete_temporary_files)

- [Avoiding GCF anti-patterns](https://cloud.google.com/blog/topics/developers-practitioners/avoiding-gcf-anti-patterns-part-2-how-reuse-cloud-function-instances-future-invocations)

- [Do lazy initialization of global variables](https://cloud.google.com/functions/docs/bestpractices/tips#do_lazy_initialization_of_global_variables)

- [Lazy globals](https://cloud.google.com/functions/docs/samples/functions-tips-lazy-globals)

- [Logs](https://cloud.google.com/functions/docs/monitoring/logging)
